homeassistant/components/notify/__init__.py

```diff
@@ -92,20 +96,49 @@ def notify_message(notify_service, call):
 
             title = template.render(
                 hass, call.data.get(ATTR_TITLE, ATTR_TITLE_DEFAULT))
-            target = call.data.get(ATTR_TARGET)
+            if targets.get(call.service) is not None:
+                target = targets[call.service]['target']
+            else:
+                target = call.data.get(ATTR_TARGET)
             message = template.render(hass, message)
             data = call.data.get(ATTR_DATA)
 
             notify_service.send_message(message, title=title, target=target,
                                         data=data)
 
         service_call_handler = partial(notify_message, notify_service)
-        service_name = slugify(p_config.get(CONF_NAME) or SERVICE_NOTIFY)
-        hass.services.register(DOMAIN, service_name, service_call_handler,
+        platform_name = (p_config.get(CONF_NAME) or SERVICE_NOTIFY)
+        platform_name_slug = slugify(platform_name)
+
+        if hasattr(notify_service, 'get_targets'):
+            for name in notify_service.get_targets().keys():
+                target_name = slugify("{}_{}".format(platform_name_slug,
+                                                     name))
+                targets[target_name] = {"platform": platform_name,
+                                        "target": name}
+                hass.services.register(DOMAIN, target_name,
+                                       service_call_handler,
+                                       descriptions.get(SERVICE_NOTIFY),
+                                       schema=NOTIFY_SERVICE_SCHEMA)
+
+        hass.services.register(DOMAIN, platform_name_slug,
+                               service_call_handler,
                                descriptions.get(SERVICE_NOTIFY),
                                schema=NOTIFY_SERVICE_SCHEMA)
         success = True
 
+    def notify_group(group, call):
```

Comment:
If we want to specify an alias that maps between targets and services, we should just do that as a _new_ notify platform. 

```yaml
notify:
  - platform: alias
    targets:
      - name: ios
        # optional
        target: robbie-i-phone
      - name: pushbullet
```